tests/utils/managed_utils/mod.rs

```diff
@@ -0,0 +1,583 @@
+#![allow(dead_code)]
+/// We allow dead code here due to the following issue:
+/// https://github.com/rust-lang/rust/issues/46379, which would otherwise create
```

Comment:
When I remove this and I run `cargo test`, I get a ton of `function is never used` warnings for functions that are used either in other functions of the framework or in functions of existing tests (that are not (yet) part of this PR). I think that's related to the issue I linked, but may have other causes. Can you do me a favor and test this on your machine? It could also just be a local issue.